src/main/java/com.vite/cli/ViteCLI.java

```diff
@@ -0,0 +1,140 @@
+package com.vite.cli;
+
+import com.vite.service.Client;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * Runs the Client
+ */
+public class ViteCLI implements CLI {
+    private final BufferedReader in;
+    private final PrintWriter out;
+    private static final Logger LOGGER = LoggerFactory.getLogger(ViteCLI.class);
+    private final Client client;
+    private final Map<String, Function<String[], String>> commands = new HashMap<>();
+    private final Map<String, String> commandDescriptions = new HashMap<>();
+    private static final String EXIT_MESSAGE = "Closing the commandline interface...";
+    private static final String HELP_MESSAGE = "Type 'help' for help";
+
+    public ViteCLI(InputStream inputStream, OutputStream outputStream, Client client) {
+        LOGGER.debug("Command line interface started...");
+        in = new BufferedReader(new InputStreamReader(inputStream));
+        out = new PrintWriter(new OutputStreamWriter(outputStream));
+        this.client = client;
+        commands.put("ping", this::ping);
+        commandDescriptions.put("ping", "Pings the server");
```

Comment:
I kinda dislike the redundancy here; i.e. that the command description has to be put into a different map, but with the same key as before.

I think either an OO approach (having a single `Map<String, Command>`, where `Command` contains all the info needed on a command), or a procedural approach (having an `addCommand(commandName, commandMethod, commandDescription)`) would be a little better.